workers/data_refinery_workers/processors/smasher.py

```diff
@@ -188,10 +210,57 @@ def _smash(job_context: Dict) -> Dict:
                 if breaker:
                     continue
 
+                # This is the innter join, the real "Smash"d
                 merged = merged.merge(frame, left_index=True, right_index=True)
 
             job_context['merged'] = merged
 
+            # Quantile Normalization
+            if job_context['dataset'].quantile_normalize:
+                try:
+                    # Prepare our QN target file
+                    organism = computed_file.samples.first().organism
+                    qn_target = utils.get_most_recent_qn_target_for_organism(organism)
+
+                    if not qn_target:
+                        logger.error("Could not find QN target for Organism!",
+                            organism=organism,
+                            dataset_id=job_context['dataset'].id,
+                            dataset_data=job_context['dataset'].data,
+                        )
+                    else:
+                        qn_target_path = qn_target.sync_from_s3()
+                        qn_target_frame = pd.read_csv(qn_target_path, sep='\t', header=None, index_col=None, error_bad_lines=False)
+
+                        # Prepare our RPy2 bridge
+                        pandas2ri.activate()
+                        preprocessCore = importr('preprocessCore')
+                        as_numeric = rlang("as.numeric")
+                        data_matrix = rlang('data.matrix')
+
+                        # Convert the smashed frames to an R numeric Matrix
+                        # and the target Dataframe into an R numeric Vector
+                        target_vector = as_numeric(qn_target_frame[0])
+                        merged_matrix = data_matrix(merged)
+
+                        # Perform the Actual QN
+                        reso = preprocessCore.normalize_quantiles_use_target(
+                                                            x=merged_matrix,
+                                                            target=target_vector,
+                                                            copy=True
+                                                        )
+
+                        # And finally convert back to Pandas
+                        ar = np.array(reso)
+                        new_merged = pd.DataFrame(ar, columns=merged.columns, index=merged.index)
+                        job_context['merged_no_qn'] = new_merged
+                        merged = new_merged
+                except Exception as e:
+                    logger.exception("Problem occured during quantile normalization",
+                        dataset_id=job_context['dataset'].id,
+                        dataset_data=job_context['dataset'].data,
```

Comment:
This is another place we'll return a un-normalized dataset without alerting the user right?